### ISSUE TITLE
Unable to import in client NextJS app

### ISSUE THREAD

**Issue by HIT2022** [ethereumjs/kzg-wasm#5]
Thanks for wrapping kzg @acolytec3 

```
`./node_modules/kzg-wasm/dist/esm/kzg.js:8:39
Module not found: Can't resolve 'module'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/kzg-wasm/dist/esm/index.js`
```

Looks to be caused by Node import in kzg.js

`if(ea){const {createRequire:a}=await import("module");var require=a(import.meta.url),fs=require("fs"),ha=require("path");`

**Comment by acolytec3** [comment]
Thanks for reporting. I've observed the same thing with Nextjs when experimenting. It does work with vite however if you are willing to use a different platform. See this repo for a working example. https://github.com/acolytec3/blobs4every1

I'll put this on my to-do list to research why it doesn't work with nextjs

**Comment by HIT2022** [comment]
My current use case is too set in NextJS so I'll have to sit tight. Appreciate the reply

**Comment by acolytec3** [comment]
I haven't figured out exactly how to resolve this yet but I can see what is happening.  The issue is that the glue code that comes with the WASM bundle does indeed include references to nodejs builtins.  That code path where those imports are invoked should never occur in a web app (which I've confirmed by local testing) but NextJS still checks those imports at compile time for some reason.  

I believe something like updating `next.config.js` so those node builtins are resolved using a polyfill or a fallback mechanism like below:
```
config = {
  webpack : {
    resolve: {
     fallback: {
       module: false,
       fs: false,
       path: false
     }
   }
  }
```

should work but my limited attempts didn't resolve it.  But, as best I can tell from reading the NextJS docs, it uses webpack under the hood for resolving and bundling up dependencies, so if you can properly set the `resolve.fallback` key in the webpack config to resolve those nodejs builtins to `false` or something similar, it should work fine.

As I noted above, `vite` doesn't seem to have this issue and I'm not an expert on `NextJS` configuration so not sure what additional help I can give.  Maybe something in [this section](https://nextjs.org/docs/pages/api-reference/next-config-js/webpack) of their docs can be of more help.

**Comment by acolytec3** [comment]
@HIT2022 see #10.  Maybe there's a way to have NextJS alias `module` to some random file that isn't loaded as in this issue where that was done for vite/vue.  I'm exploring a better fix that might have life easier for library consumers building for the browser context but haven't got it fully sketched out just yet.

**Comment by remarcable** [comment]
Hi there!

@acolytec3 the fix for Next.js works when you spread the values in your config:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  webpack: (config) => {
    return {
      ...config,
      resolve: {
        ...config.resolve,
        fallback: {
          ...config.resolve.fallback,
          module: false,
          fs: false,
          path: false,
        },
      },
    };
  },
};
module.exports = nextConfig;
```

Next.js was also complaining about not finding certain modules which then had to be installed:
```
npm i encoding pino-pretty
```

This already got me to 80%, because it worked very well in development. But sadly once I've run `npm run build`, there is an error that I couldn't fix or find the root cause for (cannot reproduce it right now).

I worked around this by copying the package from `node_modules` into my project and [commenting out](https://github.com/ephema/blob-creator/commit/ceee1b6c0c2fb6b3697fb7be4a332a421669cbcb) line `27-49` in `kzg-wasm/src/kzg.js` which were problematic. You can find an example here: https://github.com/ephema/blob-creator/tree/main/src/lib/external/kzg-wasm

I hope this helps to work around this issue, but of course this isn't a perfect solution yet.

**Comment by acolytec3** [comment]
@remarcable @HIT2022 Can you try updating your `kzg-wasm` dependency to this `git://github.com/ethereumjs/kzg-wasm#b978927` and  then run `npm i` to install that git version of the dep and see if this resolves your issues?  I'm working on a PR that adds a `browser` key to the exports map with a browser specific build of the wrapper code that removes the references to node builtins and it seems to resolve any build issues with a local NextJS app that I experimented with.

**Comment by acolytec3** [comment]
I've just released v0.4.0 of this to npm and hopefully this will resolve your import issues.  Going to close for now.
